src/Illuminate/Foundation/Console/EnvironmentDecryptCommand.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace Illuminate\Foundation\Console;
+
+use Exception;
+use Illuminate\Console\Command;
+use Illuminate\Encryption\Encrypter;
+use Illuminate\Filesystem\Filesystem;
+use Illuminate\Support\Env;
+use Illuminate\Support\Str;
+use Symfony\Component\Console\Attribute\AsCommand;
+
+#[AsCommand(name: 'env:decrypt')]
+class EnvironmentDecryptCommand extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'env:decrypt
+                    {--key= : The encryption key}
+                    {--cipher= : The encryption cipher}
+                    {--env= : The environment to be decrypted}
+                    {--force : Overwrite existing environment file}
+                    {--filename : Filename to which to write the decrypted contents}';
+
+    /**
+     * The name of the console command.
+     *
+     * This name is used to identify the command during lazy loading.
+     *
+     * @var string|null
+     *
+     * @deprecated
+     */
+    protected static $defaultName = 'env:decrypt';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Decrypt an environment file';
+
+    /**
+     * The filesystem instance.
+     *
+     * @var \Illuminate\Filesystem\Filesystem
+     */
+    protected $files;
+
+    public function __construct(Filesystem $files)
+    {
+        parent::__construct();
+
+        $this->files = $files;
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $key = $this->option('key') ?: Env::get('LARAVEL_ENV_ENCRYPTION_KEY');
```

Comment:
Supplying secrets in the command itself is considered bad practise/insecure, because they can/will be logged (e.g. to `.bash_history`). Therefore, I think there should be a way of entering the encryption key by asking the terminal for input, for example using `$this->secret()`.

Reply:
Wouldn’t it be better to allow for env and prompting and not via option. It prevents people from leaking their secret information and better UX (instead of exiting with an error if none is supplied).